src/tests/apps.cpp

```diff
@@ -55,6 +55,35 @@ void annotatedTextSentences(Ptr<Options> options, bool source) {
   }
 }
 
+void qualityEstimatorWords(const Ptr<Options> &options) {
+  ResponseOptions responseOptions;
+  const Response response = translateFromStdin(options, responseOptions);
+
+  for (const auto &wordsQualityEstimate : response.qualityScores) {
+    std::cout << "[SentenceBegin]\n";
+
+    for (const auto &wordByteRange : wordsQualityEstimate.wordByteRanges) {
+      const string_view word(response.target.text.data() + wordByteRange.begin, wordByteRange.size());
+      std::cout << word << "\n";
+    }
+    std::cout << "[SentenceEnd]\n\n";
+  }
+}
+
+void qualityEstimatorScores(const Ptr<Options> &options) {
+  ResponseOptions responseOptions;
+  const Response response = translateFromStdin(options, responseOptions);
+
+  for (const auto &wordsQualityEstimate : response.qualityScores) {
+    std::cout << std::fixed << std::setprecision(3) << wordsQualityEstimate.sentenceScore << "\n";
+
+    for (const auto &wordScore : wordsQualityEstimate.wordQualityScores) {
```

Comment:
`wordsQualityEstimate.wordQualityScores` is where you should realize the syntax offered needs to change.

Reply:
Commit c0c1bbc